### PR TITLE
[v10] Backport #12733 (tsh: Also create a PuTTY PPK-formatted keypair on login)

### PR DESCRIPTION
**Pull request by webvictim**
Backports #12733 to `branch/v10`

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

**Comment by zmb3** [comment]
@webvictim do you want to wait until my flaky test fix is merged to master and include that commit in this backport as well?

**Comment by webvictim** [comment]
Sounds good @zmb3, will do.

**Review by zmb3 — APPROVED** [review]
I've disabled auto merge so you can wait for the test fix.

**Review by Joerger — APPROVED** [review]

**Pull request merged**
